src/utils/sformat.fs

```diff
@@ -865,37 +865,74 @@ namespace Microsoft.FSharp.Text.StructuredFormat
+                                                        // returns a string, we turn off auto-quoting and esaping of
+                                                        // the string, i.e. just treat the string as display text.
+                                                        // This allows simple implementations of 
+                                                        // such as
+                                                        //
+                                                        //    [<StructuredFormatDisplay("{StructuredDisplayString}I")>]
+                                                        //    type BigInt(signInt:int, v : BigNat) =
+                                                        //        member x.StructuredDisplayString = x.ToString()
+                                                        //
+                                                        | :? string as s -> sepL s
+                                                        | _ -> 
+                                                          // recursing like this can be expensive, so let's throttle it severely
+                                                          sameObjL (depthLim/10) Precedence.BracketIfTuple alternativeObj
+                                                  countNodes 0 // 0 means we do not count the preText and postText 
+
+                                                  let postTextMatch = System.Text.RegularExpressions.Regex.Match(postText, messageRegexPattern)
+                                                  // the postText for this node will be everything up to the next occurrence of an opening brace, if one exists
+                                                  let currentPostText =
+                                                    match postTextMatch.Success with
+                                                      | false -> postText 
+                                                      | true -> replaceEscapedBrackets(postTextMatch.Groups.get_Item(1).Value)
+
+                                                  let newLayouts = (sepL preText ^^ alternativeObjL ^^ sepL currentPostText)::layouts
+                                                  match postText with
+                                                    | "" ->
+                                                      //We are done, build a space-delimited layout from the collection of layouts we've accumulated
+                                                      Some (spaceListL (List.rev newLayouts))
+                                                    | remainingPropertyText when postTextMatch.Success ->
+                                                      // More to process, keep going, using the postText starting at the next instance of a '{'
+                                                      let openingBracketIndex = remainingPropertyText.IndexOf(postTextMatch.Groups.get_Item(2).Value)-1
```

Comment:
I think you already have access to the index via `postTextMatch.Groups.[2].Index`